### PR TITLE
Feature: Add script for tab completion

### PR DESCRIPTION
**Pull request by barryskal**
This PR updates some functionality of the base commands of dnit to allow tab completion to be added. 

The approach used for tab completion was basically a copy of how doit does it. There's a hidden command (`doit tabcompletion`) that just outputs a script that can be sourced in the bash terminal that looks for the relevant dodo file and adds the output of the list command to the available commands to complete. 

As such I made the following modifications to dnit:

1. Added the `--quiet` flag to the `list` command which just outputs the available commands without the descriptions. This was needed as the input to the bash tab completion script
2. Added the `tabcompletion` command that just outputs the tab completion script.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +222,17 @@ Note that the other directories can contain (non-deno) typescript project(s) and
...
+
+For those using the tool under bash, simply execute 
+
+``
+dnit tabcompletion
+``
+
+This will output a bash completion script that can be sourced in the current shell to provide tab completion for the
+relevant dnit tasks.
+
```
Comment (paul-thompson-helix) [inline README.md:235]:
```suggestion

or run `source <(dnit tabcompletion)` to generate and source the tab completion in current shell.

```

File: dnit.ts [hunk 2]
```diff
@@ -527,16 +527,45 @@ export function task(taskParams: TaskParams): Task {
...
+    );
+  }
+}
+
+function echoBashCompletionScript() {
+  console.log("# bash completion for dnit\n" +
+    "# auto-generate by `dnit tabcompletion`\n" +
+    "\n" +
+    "# to activate it you need to 'source' the generate script\n" +
+    "# $ source <generated-script>\n" +
```
Comment (paul-thompson-helix) [inline dnit.ts]:
```suggestion
    "# to activate it you need to 'source' the generated script\n" +
    "# $ source <(dnit tabcompletion)\n" +
```

**Comment by paul-thompson-helix** [comment]
Thanks Barry.  Works a treat.

Added suggestions that it can be sourced in oneliner:
```
source <(dnit tabcompletion)
```

**Comment by paul-thompson-helix** [comment]
Worth noting also that `dnit arg` parses the commandline to get that one first arg to dispatch to the intended task, which is what this PR does tab completion on.

Tasks themselves might also pull in other commandline args (either using the provided `ctx.args` or getting them from `Deno.args` directly).

**Review by PaulThompson — APPROVED** [review]

**Pull request merged**